src/components/IndexTable/components/Checkbox/Checkbox.tsx

```diff
@@ -30,22 +30,26 @@ export const Checkbox = memo(function Checkbox() {
 
   const Wrapper = condensed ? Fragment : CheckboxWrapper;
 
+  const handleOnChange = (
+    _newChecked: boolean,
+    _id: string,
+    event: React.KeyboardEvent | React.MouseEvent<Element, MouseEvent>,
+  ) => {
+    onInteraction && onInteraction(event);
+  };
+
   return (
     <Wrapper>
       <div className={styles.TableCellContentContainer}>
-        <div
-          className={wrapperClassName}
-          onClick={onInteraction}
-          onKeyUp={onInteraction}
-          onChange={stopPropagation}
```

Comment:
We want to avoid doing this as this is event bubbling management and isn't required.  Instead, we can simply subscribe to the `onChange` prop of the `PolarisCheckbox`